pyproject.toml

```
@@ -1,19 +1,112 @@
 [project]
-dynamic = ["version", "dependencies", "optional-dependencies", "urls", "classifiers", "keywords", "authors", "requires-python", "license", "readme"]
 name = "jaxsim"
+dynamic = ["version"]
+requires-python = ">= 3.10"
+description = "A differentiable physics engine and multibody dynamics library for control and robot learning."
+authors = [
+    { name = "Diego Ferigo", email = "dgferigo@gmail.com" },
+]
+maintainers = [
+    { name = "Diego Ferigo", email = "dgferigo@gmail.com" },
+    { name = "Filippo Luca Ferretti", email = "filippo.ferretti@iit.it" },
+]
+license.file = "LICENSE"
+keywords = [
+    "physics",
+    "physics engine",
+    "jax",
+    "rigid body dynamics",
+    "featherstone",
+    "reinforcement learning",
+    "robot",
+    "robotics",
+    "sdf",
+    "urdf",
+]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Framework :: Robot Framework",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: POSIX :: Linux",
+    "Operating System :: MacOS",
+    "Operating System :: Microsoft",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Games/Entertainment :: Simulation",
+    "Topic :: Scientific/Engineering :: Artificial Intelligence",
+    "Topic :: Scientific/Engineering :: Physics",
+    "Topic :: Software Development",
+]
+dependencies = [
+    "coloredlogs",
+    "jax >= 0.4.13",
+    "jaxlib >= 0.4.13",
+    "jaxlie >= 1.3.0",
+    "jax_dataclasses >= 1.4.0",
+    "pptree",
+    "rod >= 0.3.0",
+    "typing_extensions ; python_version < '3.12'",
+]
+
+[project.optional-dependencies]
+style = [
+    "black[jupyter] ~= 24.0",
+    "isort",
+    "pre-commit",
```

Do we want to add also `ruff`? It is not necessary as it is checked in CI and in `pre-commit` if used, but it can be useful
```suggestion
    "ruff",
    "pre-commit",
```